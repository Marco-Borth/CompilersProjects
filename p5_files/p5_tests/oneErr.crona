a:int;
b:bool;
fn:void(){
	a = b;
	write true + true;
	write true - false;
	write true * false;
	write false / true;
	write -false;
	arr : int array[10];

	write(a == b);
	write(a != b);
	write(a > b);
	write(a <= b);
	write(a >= b);
	write(a < b);

	write(1 && 1);
	write(1 || 0);
	write(!0);

	if(1){}
	else{}
	if(0){}
	while(1){}

	write arr;
	write fn;
	write 5 + true;
	read fn;

	write(a[2]);
	write(arr[true]);
	a();
}

newFunc : bool() {
	return 10;
	return;
}

paramFunc : bool(x : int){}

otherFunc : void() {
 write otherFunc();
 write newFunc();
 return otherFunc();
 a();
 b();
 paramFunc(true, false);
 paramFunc(1);
 paramFunc(true);
 b = newFunc();
 switch : bool;
 write (switch == paramFunc(5));
}
