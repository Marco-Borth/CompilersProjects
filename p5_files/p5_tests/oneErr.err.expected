FATAL [4,4]: Invalid assignment operation
FATAL [5,8]: Arithmetic operator applied to invalid operand
FATAL [5,15]: Arithmetic operator applied to invalid operand
FATAL [6,8]: Arithmetic operator applied to invalid operand
FATAL [6,15]: Arithmetic operator applied to invalid operand
FATAL [7,8]: Arithmetic operator applied to invalid operand
FATAL [7,15]: Arithmetic operator applied to invalid operand
FATAL [8,8]: Arithmetic operator applied to invalid operand
FATAL [8,16]: Arithmetic operator applied to invalid operand
FATAL [9,9]: Arithmetic operator applied to invalid operand
FATAL [12,10]: Invalid equality operation
FATAL [13,10]: Invalid equality operation
FATAL [14,12]: Relational operator applied to non-numeric operand
FATAL [15,13]: Relational operator applied to non-numeric operand
FATAL [16,13]: Relational operator applied to non-numeric operand
FATAL [17,12]: Relational operator applied to non-numeric operand
FATAL [19,8]: Logical operator applied to non-bool operand
FATAL [19,13]: Logical operator applied to non-bool operand
FATAL [20,8]: Logical operator applied to non-bool operand
FATAL [20,13]: Logical operator applied to non-bool operand
FATAL [21,9]: Logical operator applied to non-bool operand
FATAL [23,5]: Non-bool expression used as an if condition
FATAL [25,5]: Non-bool expression used as an if condition
FATAL [26,8]: Non-bool expression used as a while condition
FATAL [28,8]: Attempt to write array
FATAL [29,8]: Attempt to output a function
FATAL [30,12]: Arithmetic operator applied to invalid operand
FATAL [31,7]: Attempt to read a function
FATAL [33,8]: Attempt to index a non-array
FATAL [34,12]: Bad index type
FATAL [35,2]: Attempt to call a non-function
FATAL [39,9]: Bad return value
FATAL [40,2]: Missing return value
FATAL [46,8]: Attempt to write void
FATAL [48,9]: Return with a value in void function
FATAL [49,2]: Attempt to call a non-function
FATAL [50,2]: Attempt to call a non-function
FATAL [51,2]: Function call with wrong number of args
FATAL [53,12]: Type of actual does not match type of formal
Type Analysis Failed
